src/gennylib/include/gennylib/PhaseLoop.hpp

```diff
@@ -102,17 +102,21 @@ class IterationChecker final {
                            phaseContext.isNop(),
                            phaseContext.get<TimeSpec, false>("SleepBefore").value_or(TimeSpec()),
                            phaseContext.get<TimeSpec, false>("SleepAfter").value_or(TimeSpec()),
-                           phaseContext.get<RateSpec, false>("Rate")) {
-        const auto rateSpec = phaseContext.get<RateSpec, false>("Rate");
+                           phaseContext.get<RateSpec, false>("GlobalRate")) {
+        const auto rateSpec = phaseContext.get<RateSpec, false>("GlobalRate");
 
+
+        std::ostringstream defaultRLName;
+        defaultRLName << phaseContext.get<std::string>("Name") << phaseContext.getPhaseNumber();
         const auto rateLimiterName =
-            phaseContext.get<std::string, false>("RateLimiterName").value_or("defaultRateLimiter");
+            phaseContext.get<std::string, false>("RateLimiterName").value_or(defaultRLName.str());
```

Comment:
1. let's prefix this with something to prevent collisions?
2. Do we really need it unique per Phase? the limiters for any phase other than the current one will always be unused
2. do we want it to be unique per thread rather per actor name? As it is, they'll be shared by all instances of the same Actor which is not what I thought the intent was.

Reply:
> let's prefix this with something to prevent collisions?

I was expecting the name of actors to be unique. What kind of collisions did you have in mind?

> Do we really need it unique per Phase? the limiters for any phase other than the current one will always be unused

I'm pretty sure we do. The "serial" version of Cris's workload could have had both `Rate` keywords on the same actor but in different phases. I moved the above code to be after the `if (rateSpec)` statement to avoid creating unused names. However, the rate limiter itself is only constructed for phases that define rateSpec, nothing goes to waste!

> do we want it to be unique per thread rather per actor name? As it is, they'll be shared by all instances of the same Actor which is not what I thought the intent was.

Hm... it isn't unique per thread? The actor name and phase number used to construct the rate limiter name should be the same for all threads in an actor. I'm not including the actorId in the name.